File: TicketOrder/src/main/java/com/entities/Hall.java
package com.entities;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Entity
@Table
@Data
@NoArgsConstructor
public class Hall implements Serializable {

    @Id
    @Column(name = "hall_id", nullable = false)
    private UUID id;

    @Basic
    @Column(nullable = false)
    @Getter
    private Integer capacity;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    @ToString.Exclude
    @OneToMany(mappedBy = "hall")
    private List<Session> sessions;

    public Hall(int capacity) {
        this.capacity = capacity;
        id = UUID.randomUUID();
    }
}

File: main/java/com/entities/Hall.kt
package com.entities

import java.io.Serializable
import javax.persistence.*

@Entity
@Table
class Hall : Serializable {

    @Basic
    @Column(name = "capacity", nullable = false)
    private var _capacity: Int?

    constructor(capacity: Int){
        _capacity = capacity
    }

    constructor(){
        _capacity = null
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "hall_id", nullable = false)
    private var id: Long? = null

    @OneToMany(mappedBy = "hall")
    private lateinit var sessions: MutableList<Session>

    val capacity get() = _capacity
}